ADAL/src/ADAuthenticationError.m

```diff
@@ -283,6 +298,34 @@ + (ADAuthenticationError *)OAuthServerError:(NSString *)protocolCode
                                 userInfo:nil];
 }
 
++ (ADAuthenticationError *)OAuthServerError:(NSString *)protocolCode
+                                description:(NSString *)description
+                                       code:(NSInteger)code
+                              correlationId:(NSUUID *)correlationId
+                                   userInfo:(NSDictionary *)userInfo
+{
+    return [self errorWithDomainInternal:ADOAuthServerErrorDomain
+                                    code:code
+                       protocolErrorCode:protocolCode
+                            errorDetails:description
+                           correlationId:correlationId
+                                userInfo:userInfo];
+}
+
++ (ADAuthenticationError *)errorFromExistingProtectionPolicyRequiredError:(ADAuthenticationError *) error
+                                                            correlationID:(NSUUID *) correlationId
+                                                                    token:(ADTokenCacheItem*) token
+{
+    NSMutableDictionary* userInfo = [error userInfo] ? [[error userInfo] mutableCopy] : [[NSMutableDictionary alloc] initWithCapacity:1];
+    [userInfo setObject:token forKey:@"ADMAMToken"];
```

Comment:
@jasoncoolmax why're we putting the token to the error userInfo?

Reply:
In Mike's current implementation, this method is only used by broker, where broker stores intune mam tokens in an error object. Such an error object will be handled differently when broker returns to caller app.